compose/ui/ui/src/jsWasmMain/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt

```diff
@@ -258,23 +266,16 @@ fun CanvasBasedWindow(
         // because the default behaviour expects that the Canvas takes the entire window space,
         // so the app has the same lifecycle as the browser tab.
         window.addEventListener("resize", { _ ->
-            val w = document.documentElement?.clientWidth ?: 0
-            val h = document.documentElement?.clientHeight ?: 0
-            channel.trySend(IntSize(w, h))
+            channel.trySend(getParentContainerBox())
         })
 
+        channel.trySend(getParentContainerBox())
```

Comment:
We didn't have this before. Is it necessary to fix the issue?  Or is it here for other reasons?

Reply:
We trigger exactly the same event the very first time as we trigger on resize - thus the logic of resolving/assigning dimensions is uniform and less prone for bugs when this logic is slightly different the very first time and on resize (this is exactly what happened before actually)

Reply:
Do we need `channel` here? It is asynchronous, and we can have 1-frame lag/blink because of that.

Can we just write:
```
        var composeWindow: ComposeWindow? = null
        window.addEventListener("resize", { _ ->
            composeWindow?.resize(newSize)
        })
        composeWindow = ComposeWindow(...
        composeWindow?.resize(newSize)
```
(we need to rewrite the public API `requestResize` as well, so it is a suggestion for the future)

Reply:
We intentionally do not react on every resize event anyway (there's a lot of them when we manually resize the browser window), because it's quite expensive. So we throttle the events anyway.

Reply:
Do we receive 1 resize event per frame, or it can be 10 events per frame? If it is the first, it shouldn't cause lags in theory.

We resize synchronously on desktop on every event. It is done intentionally, as the main relayout shouldn't be expensive - it just relayouts the top layout, and doesn't relayout the children. We strive to make resize as fast as the window resize, so users don't see black/white bars. We also have multiple reports that our resize still isn't ideal (because of AWT implementation we still have white bars).

As for resize on init - is there any reason to not make it synchronous? It isn't visible to user comparing to desktop (because the source code is loaded asynchronously), but I would still do that to avoid passing zero size on the first recomposition/layout - it can cause races with other init code.